packages/blade/src/components/Button/BaseButton/StyledBaseButton.native.tsx

```diff
@@ -0,0 +1,56 @@
+import styled from 'styled-components/native';
+import getBaseButtonStyles from './getBaseButtonStyles';
+import type { StyledBaseButtonProps } from './StyledBaseButton';
+
+const StyledPressable = styled.Pressable(
+  (props: Omit<StyledBaseButtonProps, 'children' | 'onClick'>) => getBaseButtonStyles(props),
+);
+
+const StyledBaseButton = ({
+  onClick,
+  children,
+  minHeight,
+  spacing,
+  isFullWidth,
+  disabled,
+  defaultBackgroundColor,
+  defaultBorderColor,
+  hoverBackgroundColor,
+  activeBackgroundColor,
+  focusBackgroundColor,
+  focusRingColor,
+  hoverBorderColor,
+  activeBorderColor,
+  focusBorderColor,
+  borderWidth,
+  borderRadius,
+}: StyledBaseButtonProps): React.ReactElement => {
+  return (
+    <StyledPressable
+      onPress={onClick}
+      style={({ pressed }): { backgroundColor: string; borderColor: string } => ({
+        backgroundColor: pressed ? activeBackgroundColor : defaultBackgroundColor,
+        borderColor: pressed ? activeBackgroundColor : defaultBackgroundColor,
```

Comment:
can't we have this directly in the styled func instead of passing it via style attribute?

Reply:
This is not the final implementation for the styling. [See this](https://github.com/razorpay/blade/pull/498/files#diff-4adb37d058188f426fa85fcce5f5ea64bcd0b62448fa8eba65cee240c606c479R37) for the final implementation